### ISSUE TITLE
Add support for shift operators for UDVT

### ISSUE THREAD

**Issue by Amxx** [ethereum/solidity#14006]
## Abstract

Solidity 0.8.19 introduced operator overloading for UDVT. However the shift operator cannot currently be overriden.

```
error[4403]: ParserError: Not a user-definable operator: >>. Only the following operators can be user-defined: |, &, ^, ~, +, -, *, /, %, ==, !=, <, >, <=, >=
```

## Specification

Allow syntax such as

```solidity
using { lshift as <<, rshift as >> } for X global;
```

with requirement that these function have type
```solidity
function lshift(X, uint256) pure returns (X); // use uint8 instead of uint256?
function rshift(X, uint256) pure returns (X); // use uint8 instead of uint256?
```

## Backwards Compatibility

No issue

**Comment by cameel** [comment]
These operators, along with `!` and `**` were deliberately excluded from the initial implementation. The blocker here is a decision about the future of implicit conversions, which has implications on type inference.

This is tied to other things we have on our roadmap and has to wait, unfortunately.

**Comment by Amxx** [comment]
Indeed `**` is also missing, and I'd argue for math library such as prb-math could definitely use it !

**Comment by cameel** [comment]
Definitely. We do want to have it eventually, it's not that we don't see the usefulness :) For now the recommended workaround is just to keep using an attached method (i.e. `exp()` added with `using for`). That's also the workaround for cases where you'd want an operator working on different types. Over time these quirks should disappear. What we delivered in 0.8.19 is just a MVP that we could do without committing ourselves to premature decisions regarding the type system.
